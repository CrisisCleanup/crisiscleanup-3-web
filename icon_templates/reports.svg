<svg xmlns="http://www.w3.org/2000/svg" width="33" height="33" viewBox="0 0 33 33">
    <path fill="#FFF" d="M19.45 7.678l.037.007.068.036 4.375 4.38v1.108l-.1.1h-.838l-.1-.1-.001-.612-3.97.002-.1-.1V8.752l-7.882-.047-.05 15.612 12.003.05v-7.88l.1-.1h.838l.1.1v8.808l-.1.1-13.879-.05-.1-.1.05-17.468.1-.1h9.448zm1.056 13.425l.1.1v.777l-.1.1h-7.053l-.1-.1v-.777l.1-.1h7.053zm.115-2.392l.1.1v.777l-.1.1h-7.097l-.1-.1v-.777l.1-.1h7.097zm-.092-2.212l.1.1v.777l-.1.1h-7.054l-.1-.1V16.6l.1-.1h7.054zm.081-2.295l.1.1v.777l-.1.1h-7.185l-.1-.1v-.777l.1-.1h7.185zm-4.426-2.419l.1.1v.777l-.1.1h-2.756l-.1-.1v-.777l.1-.1h2.756zm3.624-2.317l.001 2.153 2.187-.001-2.188-2.152z"/>
</svg>
