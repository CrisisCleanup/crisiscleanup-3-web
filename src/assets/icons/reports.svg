<svg xmlns="http://www.w3.org/2000/svg" width="33" height="33" viewBox="0 0 33 33"><path fill="#FFF" fill-rule="evenodd" d="M19.45 7.778l.034.014 4.346 4.31-.05 1.107-.838-.05.014-.661h-4.034V8.655l-8.083-.049-.05 15.813 12.153.05.05-7.982.838.05-.05 8.76-13.828-.05.05-17.468h9.447zm1.056 13.425v.777h-7.053v-.777h7.053zm.115-2.392v.777h-7.097v-.777h7.097zm-.092-2.212v.777h-7.054V16.6h7.054zm.081-2.295v.777h-7.185v-.777h7.185zm-4.426-2.419v.777h-2.756v-.777h2.756zm6.029-.164l-2.504-2.485v2.485h2.504z"/></svg>