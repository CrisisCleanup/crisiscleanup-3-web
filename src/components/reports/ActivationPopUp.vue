<template>
  <div>
    <base-button
      :text="$t('usersVue.request_access')"
      variant="solid"
      class="px-3 py-1"
      :action="() => (showRequestAccessModal = true)"
    />
    <modal
      v-if="showRequestAccessModal"
      title="Reports Activation"
      modal-classes="w-108"
      @close="showRequestAccessModal = false"
    >
      <div class="flex justify-around m-5">
        <base-text variant="body">
          Your state does not yet have a subscription to this section.
          <a
            href="mailto:aaron@crisiscleanup.org"
            class="text-crisiscleanup-yellow-900 underline"
            >Click here</a
          >
          to request more information on report access"
        </base-text>
      </div>
      <div slot="footer" class="flex p-1 justify-center">
        <base-button
          variant="outline"
          :action="() => (showRequestAccessModal = false)"
        ></base-button>
      </div>
    </modal>
  </div>
</template>

<script>
import VueTypes from 'vue-types';

export default {
  name: 'ActivationPopUp',
  props: {
    showRequestAccessModal: VueTypes.bool.def(false),
  },
};
</script>
