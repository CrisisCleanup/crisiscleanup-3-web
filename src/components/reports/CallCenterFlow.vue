<template>
  <div>
    <div class="flex flex-col justify-around">
      <div class="m-5">
        <div class="flex flex-row items-center">
          <base-text variant="body" weight="700"
            >Call Center Flow Report</base-text
          >
          <img src="@/assets/greylockss.jpg" class="h-8 px-3" />
          <!-- <img src="@/assets/yellowstarss.jpg" class="ml-auto h-8" /> -->
        </div>
        <base-text variant="bodysm">
          {{
            $t(
              '~~This is an overview of the number of calls coming into the call center each day broken down by new or follow up calls, return call rates, and volunteer operator performance.',
            )
          }}
        </base-text>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'CallCenterFlow',
};
</script>
