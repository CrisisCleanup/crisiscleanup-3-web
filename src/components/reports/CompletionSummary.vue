<template>
  <div>
    <div class="flex flex-col justify-around">
      <div class="m-5">
        <div class="flex flex-row items-center">
          <base-text variant="body" weight="700"
            >Completion Rates Summary</base-text
          >
          <img src="@/assets/greylockss.jpg" class="h-8 px-3" />
          <!-- <img src="@/assets/yellowstarss.jpg" class="ml-auto h-8" /> -->
        </div>
        <base-text variant="bodysm">
          {{
            $t(
              '~~This report breaks down the completion rate numbers by Shortest, Average, and Longest wait times for completion over all, and by county.',
            )
          }}
        </base-text>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'CompletionSummary',
};
</script>
