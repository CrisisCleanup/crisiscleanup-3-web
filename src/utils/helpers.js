/**
 * Convert rem to pixels.
 * @param rem
 * @returns {number}
 */
export const convertRemToPixels = (rem) => {
  return rem * parseFloat(getComputedStyle(document.documentElement).fontSize);
};

/**
 * Simple formatting for amounts.
 * @param num
 * @returns {number}
 */
export function nFormatter(num) {
  if (!num) return 0;
  if (num >= 1000000000) {
    return `${(num / 1000000000).toFixed(1).replace(/\.0$/, '')}G`;
  }
  if (num >= 1000000) {
    return `${(num / 1000000).toFixed(1).replace(/\.0$/, '')}M`;
  }
  if (num >= 1000) {
    return `${(num / 1000).toFixed(1).replace(/\.0$/, '')}K`;
  }
  return num;
}
