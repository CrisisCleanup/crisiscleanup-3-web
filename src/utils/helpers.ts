import { useI18n } from 'vue-i18n';
import { i18n } from '../main';

/**
 * Convert rem to pixels.
 * @param rem
 * @returns {number}
 */
export const convertRemToPixels = (rem: number): number => {
  return (
    rem * Number.parseFloat(getComputedStyle(document.documentElement).fontSize)
  );
};

/**
 * Simple formatting for amounts.
 * @returns {number | string}
 * @param number_
 */
export function nFormatter(number_: number): number | string {
  if (!number_) return 0;
  if (number_ >= 1_000_000_000) {
    return `${(number_ / 1_000_000_000).toFixed(1).replace(/\.0$/, '')}G`;
  }

  if (number_ >= 1_000_000) {
    return `${(number_ / 1_000_000).toFixed(1).replace(/\.0$/, '')}M`;
  }

  if (number_ >= 1000) {
    return `${(number_ / 1000).toFixed(1).replace(/\.0$/, '')}K`;
  }

  return number_;
}

export function formatCmsItem(text: string): string {
  return text.replaceAll(/{(.*?)}/g, (m, translation) => {
    return i18n.global.t(translation);
  });
}

export function numeral(
  value: number,
  type: string | undefined = null,
): string {
  let formatter = new Intl.NumberFormat('en-US', {
    maximumFractionDigits: 1,
    minimumFractionDigits: 0,
  });
  if (type === 'currency') {
    formatter = new Intl.NumberFormat('en-US', {
      style: 'currency',
      currency: 'USD',
      maximumFractionDigits: 0,
      minimumFractionDigits: 0,
    });
  }

  if (type === 'percentage') {
    formatter = new Intl.NumberFormat('en-US', {
      style: 'percent',
      minimumFractionDigits: 0,
      maximumFractionDigits: 0,
    });
  }

  return formatter.format(value || 0);
}
