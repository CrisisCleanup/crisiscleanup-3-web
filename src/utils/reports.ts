import moment from 'moment';

/**
 * Group objects by property.
 * `nestGroupsBy` helper method.
 *
 * @param {String} property
 * @param {Object[]} conversions
 * @returns {Object}
 */
function groupBy(conversions, property) {
  return conversions.reduce((acc, obj) => {
    const key = obj[property];
    if (!acc[key]) {
      acc[key] = [];
    }
    acc[key].push(obj);
    return acc;
  }, {});
}

function nestGroupsBy(arr, properties) {
  const props = Array.from(properties);
  if (props.length === 1) {
    return groupBy(arr, Array.from(props)[0]);
  }
  const property = props.shift();
  const grouped = groupBy(arr, property);
  // eslint-disable-next-line no-restricted-syntax
  for (const key of Object.keys(grouped)) {
    grouped[key] = nestGroupsBy(grouped[key], Array.from(props));
  }
  return grouped;
}

export function transformWidgetData(graphValue) {
  let data = {};
  if (graphValue.definition.type === 'pie') {
    const value1 = nestGroupsBy(
      graphValue.data,
      graphValue.definition.group_by,
    );
    const result = {};
    if (graphValue.data.length > 0) {
      Object.entries(value1).forEach(([wt, d]) => {
        result[wt] = [];
        Object.entries(d as any).forEach(([key, state_value]: any) => {
          const obj = state_value[0];

          Object.keys(obj).forEach((k) => {
            if (!graphValue.definition.group_by.includes(k)) {
              result[wt].push({
                name: `${key}_${k}`,
                value: obj[k],
              });
            }
          });
        });
      });
    }

    data = {
      data: result,
      group_by: graphValue.definition.group_by[0],
      type: graphValue.definition.type,
      display_options: graphValue.display_options,
    };
  } else if (graphValue.definition.type === 'multiline') {
    data = {
      data: graphValue.data,
      group_by: graphValue.definition.group_by[0],
      display_options: graphValue.display_options,
      type: graphValue.definition.type,
    };
  } else if (graphValue.definition.type === 'barstack') {
    data = {
      data: graphValue.data.map((entry) => {
        return {
          ...entry,
          [graphValue.definition.group_by[0]]: moment(
            entry[graphValue.definition.group_by[0]],
          ).toDate(),
        };
      }),
      group_by: graphValue.definition.group_by[0],
      type: graphValue.definition.type,
      display_options: graphValue.display_options,
    };
  }
  return data;
}

export function transformGraphData(rawGraphData) {
  const transformedGraphData = {};
  if (rawGraphData) {
    rawGraphData.forEach(([graphKey, graphValue]) => {
      transformedGraphData[graphKey] = transformWidgetData(graphValue);
    });
  }
  return transformedGraphData;
}
