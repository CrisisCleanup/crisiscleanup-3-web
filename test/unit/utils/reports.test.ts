import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import reportsGraphData from '../fixtures/getReportsGraphData.json';
import { MOCK_DATE } from '../../helpers';
import { transformGraphData, transformWidgetData } from '@/utils/reports';
import type { ReportWidgetGraphData } from '@/utils/reports';

describe('utils > reports', () => {
  beforeEach(() => {
    // tell vitest we use mocked time
    vi.useFakeTimers();
  });

  afterEach(() => {
    // restoring date after each test run
    vi.useRealTimers();
  });

  const d = Object.entries(reportsGraphData) as unknown as Array<
    [string, ReportWidgetGraphData<Record<string, any>>]
  >;

  for (const [graphName, graphValue] of d) {
    test(`transformWidgetData: ${graphName}`, () => {
      vi.setSystemTime(MOCK_DATE);
      const result = transformWidgetData(graphValue);
      expect(result).toMatchSnapshot();
    });
  }

  test('transformGraphData', () => {
    vi.setSystemTime(MOCK_DATE);
    const result = transformGraphData(d);
    expect(result).toMatchSnapshot();
  });
});
