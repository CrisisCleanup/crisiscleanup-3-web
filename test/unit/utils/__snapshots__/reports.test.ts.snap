// Vitest Snapshot v1

exports[`utils > reports > transformGraphData 1`] = `
{
  "cases_new_claimed_closed_cumulative_value_widget": {
    "data": [
      {
        "the_date": "2023-02-07",
        "today_claimed_case_value_cumsum": 0,
        "today_closed_case_value_cumsum": 0,
        "today_created_case_value_cumsum": null,
      },
      {
        "the_date": "2023-02-08",
        "today_claimed_case_value_cumsum": 0,
        "today_closed_case_value_cumsum": 0,
        "today_created_case_value_cumsum": null,
      },
      {
        "the_date": "2023-02-09",
        "today_claimed_case_value_cumsum": 0,
        "today_closed_case_value_cumsum": 0,
        "today_created_case_value_cumsum": null,
      },
      {
        "the_date": "2023-02-21",
        "today_claimed_case_value_cumsum": 0,
        "today_closed_case_value_cumsum": 0,
        "today_created_case_value_cumsum": null,
      },
      {
        "the_date": "2023-02-22",
        "today_claimed_case_value_cumsum": 0,
        "today_closed_case_value_cumsum": 0,
        "today_created_case_value_cumsum": null,
      },
      {
        "the_date": "2023-02-24",
        "today_claimed_case_value_cumsum": 0,
        "today_closed_case_value_cumsum": 0,
        "today_created_case_value_cumsum": null,
      },
      {
        "the_date": "2023-02-27",
        "today_claimed_case_value_cumsum": 3750,
        "today_closed_case_value_cumsum": 375,
        "today_created_case_value_cumsum": -1150,
      },
      {
        "the_date": "2023-02-28",
        "today_claimed_case_value_cumsum": 3750,
        "today_closed_case_value_cumsum": 375,
        "today_created_case_value_cumsum": -1150,
      },
    ],
    "display_options": {
      "axes": {
        "x": {
          "name": "reports.labels.date",
        },
        "y": {
          "name": "reports.labels.value_of_work",
        },
      },
      "colors": {
        "reports.cases_new_claimed_closed_cumulative_value_widget.today_claimed_case_value_cumsum": "#FAB92E",
        "reports.cases_new_claimed_closed_cumulative_value_widget.today_closed_case_value_cumsum": "#0FA355",
        "reports.cases_new_claimed_closed_cumulative_value_widget.today_created_case_value_cumsum": "#D0021B",
      },
      "number_format": "currency",
    },
    "group_by": "the_date",
    "type": "multiline",
  },
  "cases_new_claimed_closed_cumulative_widget": {
    "data": [
      {
        "the_date": "2023-02-07",
        "today_claimed_case_count_cumsum": 0,
        "today_closed_case_count_cumsum": -2.7037,
        "today_created_case_count_cumsum": 0,
      },
      {
        "the_date": "2023-02-08",
        "today_claimed_case_count_cumsum": 0,
        "today_closed_case_count_cumsum": -8.0626,
        "today_created_case_count_cumsum": 0,
      },
      {
        "the_date": "2023-02-09",
        "today_claimed_case_count_cumsum": 0,
        "today_closed_case_count_cumsum": -9.0626,
        "today_created_case_count_cumsum": 0,
      },
      {
        "the_date": "2023-02-21",
        "today_claimed_case_count_cumsum": 0,
        "today_closed_case_count_cumsum": -9.0626,
        "today_created_case_count_cumsum": 0,
      },
      {
        "the_date": "2023-02-22",
        "today_claimed_case_count_cumsum": 0,
        "today_closed_case_count_cumsum": -9.0626,
        "today_created_case_count_cumsum": 0,
      },
      {
        "the_date": "2023-02-24",
        "today_claimed_case_count_cumsum": 0,
        "today_closed_case_count_cumsum": -9.0626,
        "today_created_case_count_cumsum": 0,
      },
      {
        "the_date": "2023-02-27",
        "today_claimed_case_count_cumsum": 1,
        "today_closed_case_count_cumsum": -4.3164,
        "today_created_case_count_cumsum": 0,
      },
      {
        "the_date": "2023-02-28",
        "today_claimed_case_count_cumsum": 1,
        "today_closed_case_count_cumsum": -4.5664,
        "today_created_case_count_cumsum": 1,
      },
    ],
    "display_options": {
      "axes": {
        "x": {
          "name": "reports.labels.date",
        },
        "y": {
          "name": "reports.labels.number_of_cases",
        },
      },
      "colors": {
        "reports.cases_new_claimed_closed_cumulative_widget.today_claimed_case_count_cumsum": "#FAB92E",
        "reports.cases_new_claimed_closed_cumulative_widget.today_closed_case_count_cumsum": "#0FA355",
        "reports.cases_new_claimed_closed_cumulative_widget.today_created_case_count_cumsum": "#D0021B",
      },
      "number_format": "standard",
    },
    "group_by": "the_date",
    "type": "multiline",
  },
  "cases_new_claimed_closed_widget": {
    "data": [
      {
        "the_date": 2023-02-07T00:00:00.000Z,
        "today_claimed_case_count": 0,
        "today_closed_case_count": -2.7037,
        "today_created_case_count": 0,
      },
      {
        "the_date": 2023-02-08T00:00:00.000Z,
        "today_claimed_case_count": 0,
        "today_closed_case_count": -5.3589,
        "today_created_case_count": 0,
      },
      {
        "the_date": 2023-02-09T00:00:00.000Z,
        "today_claimed_case_count": 0,
        "today_closed_case_count": -1,
        "today_created_case_count": 0,
      },
      {
        "the_date": 2023-02-21T00:00:00.000Z,
        "today_claimed_case_count": 0,
        "today_closed_case_count": 0,
        "today_created_case_count": 0,
      },
      {
        "the_date": 2023-02-22T00:00:00.000Z,
        "today_claimed_case_count": 0,
        "today_closed_case_count": 0,
        "today_created_case_count": 0,
      },
      {
        "the_date": 2023-02-24T00:00:00.000Z,
        "today_claimed_case_count": 0,
        "today_closed_case_count": 0,
        "today_created_case_count": 0,
      },
      {
        "the_date": 2023-02-27T00:00:00.000Z,
        "today_claimed_case_count": 1,
        "today_closed_case_count": 4.7462,
        "today_created_case_count": 0,
      },
      {
        "the_date": 2023-02-28T00:00:00.000Z,
        "today_claimed_case_count": 0,
        "today_closed_case_count": -0.25,
        "today_created_case_count": 1,
      },
    ],
    "display_options": {
      "axes": {
        "x": {
          "name": "reports.labels.date",
        },
        "y": {
          "name": "reports.labels.number_of_cases",
        },
      },
      "colors": {
        "reports.cases_new_claimed_closed_widget.today_claimed_case_count": "#FAB92E",
        "reports.cases_new_claimed_closed_widget.today_closed_case_count": "#0FA355",
        "reports.cases_new_claimed_closed_widget.today_created_case_count": "#D0021B",
      },
      "number_format": "standard",
    },
    "group_by": "the_date",
    "type": "barstack",
  },
  "velocity_widget": {
    "data": [
      {
        "the_date": "2023-02-01",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-02",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-03",
        "velocity": 0.0001157315317791553,
      },
      {
        "the_date": "2023-02-04",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-05",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-06",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-07",
        "velocity": 0.006683495960246219,
      },
      {
        "the_date": "2023-02-08",
        "velocity": 0.003978271404908464,
      },
      {
        "the_date": "2023-02-09",
        "velocity": 0.001121149214110567,
      },
      {
        "the_date": "2023-02-10",
        "velocity": 0.00005786576588957765,
      },
      {
        "the_date": "2023-02-11",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-12",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-13",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-14",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-15",
        "velocity": 0.0035298117192642366,
      },
      {
        "the_date": "2023-02-16",
        "velocity": 0.0005207918930061989,
      },
      {
        "the_date": "2023-02-17",
        "velocity": 0.0030017866055218408,
      },
      {
        "the_date": "2023-02-18",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-19",
        "velocity": 0,
      },
      {
        "the_date": "2023-02-20",
        "velocity": 0.00008679864883436647,
      },
      {
        "the_date": "2023-02-21",
        "velocity": 0.003891472756074097,
      },
      {
        "the_date": "2023-02-22",
        "velocity": 0.0034864123948470534,
      },
      {
        "the_date": "2023-02-23",
        "velocity": 0.0010198841238038062,
      },
      {
        "the_date": "2023-02-24",
        "velocity": 0.002267614700797824,
      },
      {
        "the_date": "2023-02-25",
        "velocity": 0.0012802800703069057,
      },
      {
        "the_date": "2023-02-26",
        "velocity": 0.0005063254515338044,
      },
      {
        "the_date": "2023-02-27",
        "velocity": 0.00017359729766873295,
      },
      {
        "the_date": "2023-02-28",
        "velocity": 0.00040506036122704356,
      },
    ],
    "display_options": {
      "axes": {
        "x": {
          "name": "reports.labels.date",
        },
        "y": {
          "name": "reports.labels.velocity",
        },
      },
      "colors": {
        "reports.velocity_widget.velocity": "#0FA355",
      },
      "number_format": "percentage",
    },
    "group_by": "the_date",
    "type": "multiline",
  },
  "work_type_widget": {
    "data": {
      "debris": [
        {
          "name": "closed_completed_value",
          "value": 3000,
        },
        {
          "name": "open_completed_value",
          "value": 0,
        },
      ],
      "mold_remediation": [
        {
          "name": "closed_completed_value",
          "value": 12000,
        },
      ],
      "muck_out": [
        {
          "name": "closed_completed_value",
          "value": 54000,
        },
        {
          "name": "open_completed_value",
          "value": 9000,
        },
      ],
      "rebuild": [
        {
          "name": "closed_completed_value",
          "value": 20000,
        },
        {
          "name": "open_completed_value",
          "value": 5000,
        },
      ],
      "tarp": [
        {
          "name": "closed_completed_value",
          "value": 900,
        },
        {
          "name": "open_completed_value",
          "value": 0,
        },
      ],
      "trees": [
        {
          "name": "closed_completed_value",
          "value": 9375,
        },
        {
          "name": "open_completed_value",
          "value": 0,
        },
      ],
    },
    "display_options": {
      "axes": null,
      "colors": {
        "reports.work_type_widget.closed_completed_value": "#0FA355",
        "reports.work_type_widget.in_progress_completed_value": "#FAB92E",
        "reports.work_type_widget.open_completed_value": "#D0021B",
      },
      "number_format": "currency",
    },
    "group_by": "work_type_key",
    "type": "pie",
  },
}
`;

exports[`utils > reports > transformWidgetData: cases_new_claimed_closed_cumulative_value_widget 1`] = `
{
  "data": [
    {
      "the_date": "2023-02-07",
      "today_claimed_case_value_cumsum": 0,
      "today_closed_case_value_cumsum": 0,
      "today_created_case_value_cumsum": null,
    },
    {
      "the_date": "2023-02-08",
      "today_claimed_case_value_cumsum": 0,
      "today_closed_case_value_cumsum": 0,
      "today_created_case_value_cumsum": null,
    },
    {
      "the_date": "2023-02-09",
      "today_claimed_case_value_cumsum": 0,
      "today_closed_case_value_cumsum": 0,
      "today_created_case_value_cumsum": null,
    },
    {
      "the_date": "2023-02-21",
      "today_claimed_case_value_cumsum": 0,
      "today_closed_case_value_cumsum": 0,
      "today_created_case_value_cumsum": null,
    },
    {
      "the_date": "2023-02-22",
      "today_claimed_case_value_cumsum": 0,
      "today_closed_case_value_cumsum": 0,
      "today_created_case_value_cumsum": null,
    },
    {
      "the_date": "2023-02-24",
      "today_claimed_case_value_cumsum": 0,
      "today_closed_case_value_cumsum": 0,
      "today_created_case_value_cumsum": null,
    },
    {
      "the_date": "2023-02-27",
      "today_claimed_case_value_cumsum": 3750,
      "today_closed_case_value_cumsum": 375,
      "today_created_case_value_cumsum": -1150,
    },
    {
      "the_date": "2023-02-28",
      "today_claimed_case_value_cumsum": 3750,
      "today_closed_case_value_cumsum": 375,
      "today_created_case_value_cumsum": -1150,
    },
  ],
  "display_options": {
    "axes": {
      "x": {
        "name": "reports.labels.date",
      },
      "y": {
        "name": "reports.labels.value_of_work",
      },
    },
    "colors": {
      "reports.cases_new_claimed_closed_cumulative_value_widget.today_claimed_case_value_cumsum": "#FAB92E",
      "reports.cases_new_claimed_closed_cumulative_value_widget.today_closed_case_value_cumsum": "#0FA355",
      "reports.cases_new_claimed_closed_cumulative_value_widget.today_created_case_value_cumsum": "#D0021B",
    },
    "number_format": "currency",
  },
  "group_by": "the_date",
  "type": "multiline",
}
`;

exports[`utils > reports > transformWidgetData: cases_new_claimed_closed_cumulative_widget 1`] = `
{
  "data": [
    {
      "the_date": "2023-02-07",
      "today_claimed_case_count_cumsum": 0,
      "today_closed_case_count_cumsum": -2.7037,
      "today_created_case_count_cumsum": 0,
    },
    {
      "the_date": "2023-02-08",
      "today_claimed_case_count_cumsum": 0,
      "today_closed_case_count_cumsum": -8.0626,
      "today_created_case_count_cumsum": 0,
    },
    {
      "the_date": "2023-02-09",
      "today_claimed_case_count_cumsum": 0,
      "today_closed_case_count_cumsum": -9.0626,
      "today_created_case_count_cumsum": 0,
    },
    {
      "the_date": "2023-02-21",
      "today_claimed_case_count_cumsum": 0,
      "today_closed_case_count_cumsum": -9.0626,
      "today_created_case_count_cumsum": 0,
    },
    {
      "the_date": "2023-02-22",
      "today_claimed_case_count_cumsum": 0,
      "today_closed_case_count_cumsum": -9.0626,
      "today_created_case_count_cumsum": 0,
    },
    {
      "the_date": "2023-02-24",
      "today_claimed_case_count_cumsum": 0,
      "today_closed_case_count_cumsum": -9.0626,
      "today_created_case_count_cumsum": 0,
    },
    {
      "the_date": "2023-02-27",
      "today_claimed_case_count_cumsum": 1,
      "today_closed_case_count_cumsum": -4.3164,
      "today_created_case_count_cumsum": 0,
    },
    {
      "the_date": "2023-02-28",
      "today_claimed_case_count_cumsum": 1,
      "today_closed_case_count_cumsum": -4.5664,
      "today_created_case_count_cumsum": 1,
    },
  ],
  "display_options": {
    "axes": {
      "x": {
        "name": "reports.labels.date",
      },
      "y": {
        "name": "reports.labels.number_of_cases",
      },
    },
    "colors": {
      "reports.cases_new_claimed_closed_cumulative_widget.today_claimed_case_count_cumsum": "#FAB92E",
      "reports.cases_new_claimed_closed_cumulative_widget.today_closed_case_count_cumsum": "#0FA355",
      "reports.cases_new_claimed_closed_cumulative_widget.today_created_case_count_cumsum": "#D0021B",
    },
    "number_format": "standard",
  },
  "group_by": "the_date",
  "type": "multiline",
}
`;

exports[`utils > reports > transformWidgetData: cases_new_claimed_closed_widget 1`] = `
{
  "data": [
    {
      "the_date": 2023-02-07T00:00:00.000Z,
      "today_claimed_case_count": 0,
      "today_closed_case_count": -2.7037,
      "today_created_case_count": 0,
    },
    {
      "the_date": 2023-02-08T00:00:00.000Z,
      "today_claimed_case_count": 0,
      "today_closed_case_count": -5.3589,
      "today_created_case_count": 0,
    },
    {
      "the_date": 2023-02-09T00:00:00.000Z,
      "today_claimed_case_count": 0,
      "today_closed_case_count": -1,
      "today_created_case_count": 0,
    },
    {
      "the_date": 2023-02-21T00:00:00.000Z,
      "today_claimed_case_count": 0,
      "today_closed_case_count": 0,
      "today_created_case_count": 0,
    },
    {
      "the_date": 2023-02-22T00:00:00.000Z,
      "today_claimed_case_count": 0,
      "today_closed_case_count": 0,
      "today_created_case_count": 0,
    },
    {
      "the_date": 2023-02-24T00:00:00.000Z,
      "today_claimed_case_count": 0,
      "today_closed_case_count": 0,
      "today_created_case_count": 0,
    },
    {
      "the_date": 2023-02-27T00:00:00.000Z,
      "today_claimed_case_count": 1,
      "today_closed_case_count": 4.7462,
      "today_created_case_count": 0,
    },
    {
      "the_date": 2023-02-28T00:00:00.000Z,
      "today_claimed_case_count": 0,
      "today_closed_case_count": -0.25,
      "today_created_case_count": 1,
    },
  ],
  "display_options": {
    "axes": {
      "x": {
        "name": "reports.labels.date",
      },
      "y": {
        "name": "reports.labels.number_of_cases",
      },
    },
    "colors": {
      "reports.cases_new_claimed_closed_widget.today_claimed_case_count": "#FAB92E",
      "reports.cases_new_claimed_closed_widget.today_closed_case_count": "#0FA355",
      "reports.cases_new_claimed_closed_widget.today_created_case_count": "#D0021B",
    },
    "number_format": "standard",
  },
  "group_by": "the_date",
  "type": "barstack",
}
`;

exports[`utils > reports > transformWidgetData: velocity_widget 1`] = `
{
  "data": [
    {
      "the_date": "2023-02-01",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-02",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-03",
      "velocity": 0.0001157315317791553,
    },
    {
      "the_date": "2023-02-04",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-05",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-06",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-07",
      "velocity": 0.006683495960246219,
    },
    {
      "the_date": "2023-02-08",
      "velocity": 0.003978271404908464,
    },
    {
      "the_date": "2023-02-09",
      "velocity": 0.001121149214110567,
    },
    {
      "the_date": "2023-02-10",
      "velocity": 0.00005786576588957765,
    },
    {
      "the_date": "2023-02-11",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-12",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-13",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-14",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-15",
      "velocity": 0.0035298117192642366,
    },
    {
      "the_date": "2023-02-16",
      "velocity": 0.0005207918930061989,
    },
    {
      "the_date": "2023-02-17",
      "velocity": 0.0030017866055218408,
    },
    {
      "the_date": "2023-02-18",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-19",
      "velocity": 0,
    },
    {
      "the_date": "2023-02-20",
      "velocity": 0.00008679864883436647,
    },
    {
      "the_date": "2023-02-21",
      "velocity": 0.003891472756074097,
    },
    {
      "the_date": "2023-02-22",
      "velocity": 0.0034864123948470534,
    },
    {
      "the_date": "2023-02-23",
      "velocity": 0.0010198841238038062,
    },
    {
      "the_date": "2023-02-24",
      "velocity": 0.002267614700797824,
    },
    {
      "the_date": "2023-02-25",
      "velocity": 0.0012802800703069057,
    },
    {
      "the_date": "2023-02-26",
      "velocity": 0.0005063254515338044,
    },
    {
      "the_date": "2023-02-27",
      "velocity": 0.00017359729766873295,
    },
    {
      "the_date": "2023-02-28",
      "velocity": 0.00040506036122704356,
    },
  ],
  "display_options": {
    "axes": {
      "x": {
        "name": "reports.labels.date",
      },
      "y": {
        "name": "reports.labels.velocity",
      },
    },
    "colors": {
      "reports.velocity_widget.velocity": "#0FA355",
    },
    "number_format": "percentage",
  },
  "group_by": "the_date",
  "type": "multiline",
}
`;

exports[`utils > reports > transformWidgetData: work_type_widget 1`] = `
{
  "data": {
    "debris": [
      {
        "name": "closed_completed_value",
        "value": 3000,
      },
      {
        "name": "open_completed_value",
        "value": 0,
      },
    ],
    "mold_remediation": [
      {
        "name": "closed_completed_value",
        "value": 12000,
      },
    ],
    "muck_out": [
      {
        "name": "closed_completed_value",
        "value": 54000,
      },
      {
        "name": "open_completed_value",
        "value": 9000,
      },
    ],
    "rebuild": [
      {
        "name": "closed_completed_value",
        "value": 20000,
      },
      {
        "name": "open_completed_value",
        "value": 5000,
      },
    ],
    "tarp": [
      {
        "name": "closed_completed_value",
        "value": 900,
      },
      {
        "name": "open_completed_value",
        "value": 0,
      },
    ],
    "trees": [
      {
        "name": "closed_completed_value",
        "value": 9375,
      },
      {
        "name": "open_completed_value",
        "value": 0,
      },
    ],
  },
  "display_options": {
    "axes": null,
    "colors": {
      "reports.work_type_widget.closed_completed_value": "#0FA355",
      "reports.work_type_widget.in_progress_completed_value": "#FAB92E",
      "reports.work_type_widget.open_completed_value": "#D0021B",
    },
    "number_format": "currency",
  },
  "group_by": "work_type_key",
  "type": "pie",
}
`;
